### PR TITLE
wip: pgbouncer

### PR DESCRIPTION
**Pull request by ntarocco**
Nice article to read: https://www.depesz.com/2012/12/02/what-is-the-point-of-bouncing/

### CODE REVIEW

File: docker/pgbouncer/pgbouncer.ini [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+;;;
+;;; Pooler personality questions
+;;;
+
+;; When server connection is released back to pool:
+;;   session      - after client disconnects (default)
+;;   transaction  - after transaction finishes
+;;   statement    - after statement finishes
+pool_mode = session
```
Comment (ntarocco) [inline docker/pgbouncer/pgbouncer.ini:127]: This parameter must be checked very carefully

File: invenio.cfg [hunk 2]
```diff
@@ -68,8 +69,19 @@ APP_RDM_ROUTES["index"] = ("/", frontpage_view_function)
 # Flask-SQLAlchemy
 # ================
 # See https://flask-sqlalchemy.palletsprojects.com/en/2.x/config/
-
-SQLALCHEMY_DATABASE_URI="postgresql+psycopg2://zenodo:zenodo@localhost/zenodo"
+def _parse_env_bool(var_name, default=None):
```
Comment (ntarocco) [inline invenio.cfg:72]: can be removed when `pgbouncer` will be enabled by default

**Review by slint — APPROVED** [review]

File: docker/pgbouncer/pgbouncer.ini [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+;;;
+;;; Pooler personality questions
+;;;
+
+;; When server connection is released back to pool:
+;;   session      - after client disconnects (default)
+;;   transaction  - after transaction finishes
+;;   statement    - after statement finishes
+pool_mode = transaction
```
Comment (ntarocco) [inline docker/pgbouncer/pgbouncer.ini:127]: Pooler/connections settings must be reviewed carefully

Comment (slint) [reply docker/pgbouncer/pgbouncer.ini:127]: Based on the article above it looks ok. I don't think we ever rely on sequence values across transactions (e.g. for the [`recid` sequence](https://github.com/inveniosoftware/invenio-pidstore/blob/d54359c185c92f87decc8b9ba951ed03790226f6/invenio_pidstore/models.py#L575) of the [requests `number`](https://github.com/inveniosoftware/invenio-requests/blob/31b4d5d99adafde6e358d6b86ae046b2885be6b2/invenio_requests/records/models.py#L129), and hopefully SQLAlchemy doesn't do any fancy tricks under the hood either.

**Pull request merged**